File: Python/baekjoon/단계별로 풀어보기/정렬/1181_단어 정렬.py
# https://www.acmicpc.net/problem/1181

from sys import stdin

input = stdin.readline
li = []
for _ in range(int(input())):
    li.append(input().rstrip())
li = sorted(list(set(li)), key=lambda x: (len(x), x))
for i in li:
    print(i)

File: C#/Baekjoon/정렬/1181_단어 정렬.cs
//https://www.acmicpc.net/problem/1181

using var sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
using var sw = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
int n = int.Parse(sr.ReadLine());
var S = new string[n];
for (int i = 0; i < n; i++)
{
    S[i] = sr.ReadLine();
}
foreach (var s in S.Distinct().OrderBy(x => x.Length).ThenBy(x => x)) sw.WriteLine(s);